R/derive_param_lasta.R

```diff
@@ -0,0 +1,358 @@
+#'   "01",     lubridate::ymd("2020-12-06"), lubridate::ymd("2022-03-06"),
+#'   "02",     lubridate::ymd("2021-01-16"), lubridate::ymd("2022-02-03"),
+#'   "03",     lubridate::ymd("2021-01-09"), lubridate::ymd("2021-02-24"),
+#'   "04",     lubridate::ymd("2021-04-21"), lubridate::ymd("2021-09-15")
+#' ) %>%
+#'   dplyr::mutate(STUDYID = "a_study_id")
+#'
+#' adrs <- tibble::tribble(
+#'   ~USUBJID, ~PARAMCD, ~AVAL, ~AVALC, ~ASEQ, ~ADT,
+#'   "01", "RSP", NA, "Y", 1, lubridate::ymd("2021-04-08"),
+#'   "02", "RSP", NA, "N", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "RSP", NA, "N", 1, NA,
+#'   "04", "RSP", NA, "N", 1, NA,
+#'   "01", "PD",  NA, "N", 1, NA,
+#'   "02", "PD",  NA, "Y", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "PD",  NA, "N", 1, NA,
+#'   "04", "PD",  NA, "N", 1, NA,
+#'   "01", "OVR", 3, "SD", 1, lubridate::ymd("2021-03-07"),
+#'   "01", "OVR", 2, "PR", 1, lubridate::ymd("2021-04-08"),
+#'   "02", "OVR", 3, "SD", 1, lubridate::ymd("2021-03-07"),
+#'   "02", "OVR", NA, NA, 1, lubridate::ymd("2021-04-07"),
+#'   "02", "OVR", 6, "PD", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "OVR", 3, "SD", 1, lubridate::ymd("2021-01-30"),
+#'   "03", "OVR", 3, "SD", 2, lubridate::ymd("2021-01-30"),
+#'   "04", "OVR", NA, "NE", 1, lubridate::ymd("2021-05-21"),
+#'   "04", "OVR", 5, "NON-PD", 1, lubridate::ymd("2021-06-30"),
+#'   "04", "OVR", NA, "NE", 1, lubridate::ymd("2021-07-24"),
+#'   "04", "OVR", NA, "ND", 1, lubridate::ymd("2021-09-30"),
+#' ) %>%
+#'   dplyr::mutate(STUDYID = "a_study_id")
+#'
+#' pd <- date_source(
+#'   dataset_name = adrs,
+#'   date         = ADT,
+#'   filter       = PARAMCD == 'PD' & AVALC == 'Y'
+#' )
+#'
+#' derive_param_lasta(
+#'   dataset,
+#'   filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
+#'   source_pd = list(pd),
+#'   source_datasets = list(adrs = adrs),
+#'   set_values_to = vars(
+#'     PARAMCD = "LSTAC",
+#'     PARAM = "Last Disease Assessment Censored at First PD by Investigator",
+#'     PARCAT1 = "Tumor Response",
+#'     PARCAT2 = "Investigator",
+#'     PARCAT3 = "Recist 1.1",
+#'     ANL01FL = "Y")
+#' )
+#'}
+#'
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @export
+#' @name   derive_param_lasta
+#' @title  derive_param_lasta
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @author Stephen Gormley
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @keywords ADRS
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @importFrom  magrittr `%>%`
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @return The dataframe passed in the `dataset` argument with additional
+#'         columns and/or rows as set in the `set_values_to` argument.
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%|
+
+derive_param_lasta <- function(dataset,
+                               dataset_adsl,
```

Comment:
I added this based on this conversation in slack, which was after the discussion we had in #25

https://pharmaverse.slack.com/archives/C032B5SP2G2/p1655118369051019

I can remove, but can you just confirm this is correct based on this slack discussion?

Reply:
I see. I think `derive_param_lasta()` is an exception because the parameter is intended for time to event analyses. Adding observations for patients without a tumor assessment makes no sense because they do not contain the required information (`ADT` is missing for those).

@rossfarrugia , do you agree?

Reply:
you can remove @sgorm123 and refer to Catherine's comment in #25 as to the reasoning